Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py

```diff
@@ -474,10 +492,19 @@ def _request(cls, url, ca_cert_path=None):
 
     @classmethod
     def enabled(cls):
+        if cls._temporary_disable > 0:
+            return False
         if os.environ.get(cls.DISABLE_ENV_VAR) not in ['0', 'FALSE', 'False', 'false', 'NO', 'No', 'no', None]:
             return False
         return True if cls.directory else None
 
+    @classmethod
+    @contextmanager
```

Comment:
Why is this a context manager? Seems like `classmethod` would suffice.

Reply:
See the `yield` statement. We want to be able to temporarily disable it, and then re-enable it afterwards.